src/icastats_shared.c

```diff
@@ -54,9 +54,11 @@ int stats_mmap(int user)
 	sprintf(shm_id, "icastats_%d",
 		user == -1 ? geteuid() : (uid_t)user);
 
-	stats_shm_handle = shm_open(shm_id,
-				    O_CREAT | O_RDWR,
-				    S_IRUSR | S_IWUSR);
+	stats_shm_handle = shm_open(shm_id, O_RDWR,  S_IRUSR | S_IWUSR);
+
+	if (stats_shm_handle == -1)
+		stats_shm_handle = shm_open(shm_id, O_CREAT | O_RDWR,
+											S_IRUSR | S_IWUSR);
```

Comment:
I would say there are too many tabs.  With a tab size of 8 (which fits to all other places) there should be only 5 tabs.

Reply:
Yep ... saw it also. In eclipse it looked ok.
Fixed.

Reply:
I think its still too many tabs....
Make sure you editor setting displays a tab as 8 characters !

Reply:
Well, I have 4 chars per tab and don't want to change this. Removing more tabs would make it look bad in eclipse. I know that for kernel development, 8 chars per tab are a must. But we never followed this strictly in libica.

Reply:
Well, the remaining libica code uses 8 chars per tab.

See a few lines above:

```
	stats_shm_handle = shm_open(shm_id,
				    O_CREAT | O_RDWR,
				    S_IRUSR | S_IWUSR);
```

This is correctly aligned with 8 chars per tab, but wrong for 4 chars per tab. 

You can set the chars per tab in eclipse specific for each project, so that it looks good for every style.

Reply:
Everything in one line now. BTW: the old code had 6 tabs plus 4 spaces, which I really don't like.

Reply:
much better now! Thanks.